stage0/src/main.rs

```diff
@@ -50,6 +52,92 @@ const ROM_VER: u32 = 0;
 #[cfg(not(feature = "0A-hardware"))]
 const ROM_VER: u32 = 1;
 
+#[cfg(feature = "secure")]
+unsafe fn branch_to_image(image: Image) -> ! {
+    let sau_ctrl: *mut u32 = 0xe000edd0 as *mut u32;
+    let sau_rbar: *mut u32 = 0xe000eddc as *mut u32;
+    let sau_rlar: *mut u32 = 0xe000ede0 as *mut u32;
+    let sau_rnr: *mut u32 = 0xe000edd8 as *mut u32;
+
+    for i in 0..8 {
+        if let Some(r) = image.get_sau_entry(i) {
+            core::ptr::write_volatile(sau_rnr, i as u32);
+            core::ptr::write_volatile(sau_rbar, r.rbar);
+            core::ptr::write_volatile(sau_rlar, r.rlar);
+        }
+    }
+
+    core::ptr::write_volatile(sau_ctrl, 1);
+
+    let mut peripherals = match Peripherals::take() {
+        Some(p) => p,
+        None => loop {},
```

Comment:
Is `loop {}` going to get compiled to something odd?  Cliff said at one point

> (`loop {}` for better or worse becomes a panic when LLVM finds it)

and suggested putting a `cortex_m::asm::nop()` in there

Reply:
What about `panic!`? The `panic-halt` crate ought to work in this context.